### PR TITLE
QA-1488: Build Cloud function streaming capability

### PR DESCRIPTION
**Pull request by ichengchang**
This purpose of this Pull Request is to review the following chuck of work related to [QA-1488](https://broadworkbench.atlassian.net/browse/QA-1488)

- Migrated cloud build steps into GitHub Action workflow for building and deploying cloud functions for `java11`
- The GitHub Action has been updated to use GitHub secrets as recommended by DevOps for the following reasons.

The secret key for publishing the Cloud Function library to Google Storage was added directly as GitHub secret. Doing so allows us to sidestep the vault-token-step to pull secrets from the vault*.

DevOps is trying to discourage pulling secrets from vault in Github Actions at runtime as it's very easy to accidentally leak them in Github Actions logs.
- Add cloud logging support so cloud function messages can be viewed in Google Logs Explorer
- Demonstrate event trigger the download of test results archive from Google Bucket, unpack archive into streams, read metadata of unpacked files and send them to cloud log

### CODE REVIEW

File: src/main/java/bio/terra/cloudevents/GCSEvent.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
+package bio.terra.cloudevents;
+
+import java.util.Date;
+
+public class GCSEvent {
```
Comment (davidangb) [inline src/main/java/bio/terra/cloudevents/GCSEvent.java:5]: Do you need to write this class yourself? Can you use `com.google.events.cloud.pubsub.v1.Message` or `com.google.events.cloud.storage.v1. StorageObjectData` or one of the other events from 
https://github.com/googleapis/google-cloudevents-java instead?

File: src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java [hunk 2]
```diff
@@ -1,83 +1,91 @@
...
+
+    // ArchiveInputStream is a special type of InputStream that emits an EOF when it gets to the end
+    // of a file in the archive.
+    // Once it’s done, call getNextEntry to reset the stream and start reading the next file.
+    // When getNextEntry returns null, you’re at the end of the archive.
+    ArchiveEntry archiveEntry;
+    while ((archiveEntry = archiveInputStream.getNextEntry()) != null) {
+      logger.info(archiveEntry.getName() + " " + archiveEntry.getSize() + " bytes");
     }
+  }
```
Comment (davidangb) [inline src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java:68]: I'd really like to separate the different pieces of code into separate classes, for encapsulation and future-proofing. For instance, we should have one "harness" class for interfacing with Cloud Functions, one or more classes for handling TestRunner files, etc. I've made a rough sketch of this here: https://lucid.app/lucidchart/invitations/accept/inv_fc1efe60-354d-4d0d-a91d-44760463a6df?viewport_loc=-44%2C12%2C2503%2C1291%2C7ezayB~FQlW7. I don't expect this refactoring as part of this PR, but thought I'd mention it.

**Review by davidangb — COMMENTED** [review]
a few comments inline. Beyond those: are there any unit tests for this code? What about error-handling/error-trapping for cases such as a failure to read the file from the bucket?

File: src/main/java/bio/terra/cloudevents/GCSEvent.java (hunk 1 above)
Comment (ichengchang) [reply src/main/java/bio/terra/cloudevents/GCSEvent.java:5]: Created story [QA-1522](https://broadworkbench.atlassian.net/browse/QA-1522) to work on this feedback.

File: src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java (hunk 2 above)
Comment (ichengchang) [reply src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java:195]: Created story [QA-1523](https://broadworkbench.atlassian.net/browse/QA-1523) to work on the comments.

**Pull request merged**